adapter/internal/oasparser/config_generator.go

```diff
@@ -196,11 +196,12 @@ func GetEnforcerAPI(mgwSwagger model.MgwSwagger, lifeCycleState string, vhost st
 		}
 	}
 
-	choreoComponentInfo := &api.ChoreoComponentInfo{
-		ComponentID:    mgwSwagger.ChoreoComponentInfo.ComponentID,
-		VersionID:      mgwSwagger.ChoreoComponentInfo.VersionID,
-		OrganizationID: mgwSwagger.ChoreoComponentInfo.OrganizationID,
-		ProjectID:      mgwSwagger.ChoreoComponentInfo.ProjectID,
+	choreoComponentInfo := &api.ChoreoComponentInfo{}
+	if mgwSwagger.ChoreoComponentInfo != nil {
```

Comment:
If this is something we expect to be mandatory for a given api, we can add a log and monitor from logs for a while.

Reply:
discussed and decided we won't require that sort of a log.

Reply:
This is not critical atm